Webflux:  issue with liquibase + PostgreSQL

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**
cannot authenticate after jhipster generation of Webflux postgreSQL project

```
org.springframework.r2dbc.BadSqlGrammarException: executeMany; bad SQL grammar [SELECT * FROM jhi_user u LEFT JOIN jhi_user_authority ua ON u.id=ua.user_id WHERE u.login = :login]; nested exception is io.r2dbc.postgresql.ExceptionFactory$PostgresqlBadGrammarException: [42P01] relation "jhi_user" does not exist
```

##### **Motivation for or Use Case**

I'd expect the /authenticate endpoint to work out of the box

##### **Reproduce the error**

Initialise Monolith WebFlux PostgreSQL

##### **Related issues**

##### **Suggest a Fix**
com.mycompany.myapp.web.rest.UserJWTController.lambda$authorize$2(UserJWTController.java:40)

##### **JHipster Version(s)**

7.1.0

##### **JHipster configuration**
##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "pgreactive",
    "jhipsterVersion": "7.1.0",
    "skipClient": true,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1631269877243,
    "serviceDiscoveryType": false,
    "reactive": true,
    "authenticationType": "jwt",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "no",
    "enableTranslation": false,
    "nativeLanguage": "en",
    "packageFolder": "com/mycompany/myapp",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "devServerPort": 9060,
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "withAdminUi": true,
    "languages": ["en", "fr"],
    "lastLiquibaseTimestamp": 1631270352000,
    "entities": ["BarProperty"]
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity BarProperty {
  legId String
  priority Integer
  receivedAt ZonedDateTime
  field String
  value String
  issuedAt ZonedDateTime
}

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)

git version 2.33.0

node: v16.8.0

npm: 7.21.0

Docker version 20.10.7, build f0df350

docker-compose version 1.29.2, build 5becea4c


No change to package.json was detected. No package manager install will be executed.

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

no entities

##### **Browsers and Operating System**

Mac OS Big Sur 11.5 - not browser related

I tried your `.yo-rc.json` and JDL with the `main` branch. It spits out the following error on startup.

```
2021-09-10 12:31:01.792 ERROR 72989 --- [reactive-task-1] t.j.c.liquibase.AsyncSpringLiquibase     : Liquibase could not start correctly, your database is NOT ready: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20210910183011_added_entity_BarProperty.xml::20210910183011-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: column "received_at" is of type timestamp without time zone but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 112

liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20210910183011_added_entity_BarProperty.xml::20210910183011-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: column "received_at" is of type timestamp without time zone but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 112
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:124)
	at liquibase.Liquibase.lambda$null$0(Liquibase.java:265)
	at liquibase.Scope.lambda$child$0(Scope.java:166)
	at liquibase.Scope.child(Scope.java:175)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:144)
	at liquibase.Scope.child(Scope.java:228)
	at liquibase.Liquibase.lambda$update$1(Liquibase.java:264)
	at liquibase.Scope.lambda$child$0(Scope.java:166)
	at liquibase.Scope.child(Scope.java:175)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:144)
	at liquibase.Liquibase.runInScope(Liquibase.java:2404)
	at liquibase.Liquibase.update(Liquibase.java:211)
	at liquibase.Liquibase.update(Liquibase.java:197)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:314)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:269)
	at org.springframework.boot.autoconfigure.liquibase.DataSourceClosingSpringLiquibase.afterPropertiesSet(DataSourceClosingSpringLiquibase.java:46)
	at tech.jhipster.config.liquibase.AsyncSpringLiquibase.initDb(AsyncSpringLiquibase.java:118)
	at tech.jhipster.config.liquibase.AsyncSpringLiquibase.lambda$afterPropertiesSet$0(AsyncSpringLiquibase.java:93)
	at tech.jhipster.async.ExceptionHandlingAsyncTaskExecutor.lambda$createWrappedRunnable$1(ExceptionHandlingAsyncTaskExecutor.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20210910183011_added_entity_BarProperty.xml::20210910183011-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: column "received_at" is of type timestamp without time zone but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 112
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:695)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:49)
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:166)
	at liquibase.Scope.child(Scope.java:175)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:144)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:110)
	at liquibase.Scope.lambda$child$0(Scope.java:166)
	at liquibase.Scope.child(Scope.java:175)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:144)
	at liquibase.Scope.child(Scope.java:228)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94)
	at liquibase.Scope.lambda$child$0(Scope.java:166)
	at liquibase.Scope.child(Scope.java:175)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:144)
	at liquibase.Scope.child(Scope.java:228)
	at liquibase.Scope.child(Scope.java:232)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66)
	... 23 common frames omitted
Caused by: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: column "received_at" is of type timestamp without time zone but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 112
	at liquibase.statement.ExecutablePreparedStatementBase.execute(ExecutablePreparedStatementBase.java:82)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:140)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1279)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1261)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:660)
	... 43 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: column "received_at" is of type timestamp without time zone but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 112
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2552)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2284)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:322)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:153)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at liquibase.statement.ExecutablePreparedStatementBase.executePreparedStatement(ExecutablePreparedStatementBase.java:95)
	at liquibase.statement.ExecutablePreparedStatementBase.execute(ExecutablePreparedStatementBase.java:80)
	... 47 common frames omitted
```

--- a/utils/liquibase.js
+++ b/utils/liquibase.js
@@ -161,7 +161,11 @@ function parseLiquibaseLoadColumnType(entity, field) {
   }
 
   // eslint-disable-next-line no-template-curly-in-string
-  if (['date', '${datetimeType}', 'boolean'].includes(columnType)) {
+  if (['date', '${datetimeType}'].includes(columnType)) {
+    return 'date';
+  }
+
+  if(columnType === 'boolean') {
     return columnType;
   }
 

--- a/utils/liquibase.js
+++ b/utils/liquibase.js
@@ -165,7 +165,7 @@ function parseLiquibaseLoadColumnType(entity, field) {
     return 'date';
   }
 
-  if(columnType === 'boolean') {
+  if (columnType === 'boolean') {
     return columnType;
   }
 

